src/record/peer-record/index.js

```diff
@@ -0,0 +1,98 @@
+'use strict'
+
+const multiaddr = require('multiaddr')
+const PeerId = require('peer-id')
+const Record = require('libp2p-interfaces/src/record')
+
+const Protobuf = require('./peer-record.proto')
+const {
+  ENVELOPE_DOMAIN_PEER_RECORD,
+  ENVELOPE_PAYLOAD_TYPE_PEER_RECORD
+} = require('./consts')
+
+const arraysAreEqual = (a, b) => a.length === b.length && a.sort().every((item, index) => b[index].equals(item))
+
+/**
+ * The PeerRecord is responsible for TODOTODOTRDO
```

Comment:
Probably don't need to define responsibility here. What it is would be helpful though.